app/xds_api/views.py

```diff
@@ -358,3 +374,114 @@ def add_course_to_lists(request, exp_hash):
     else:
         return Response({"message": "course successfully added!"},
                         status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+def interest_lists_owned(request):
+    """Handles HTTP requests for interest lists managed by request user"""
+    errorMsg = {
+        "message": "Error fetching records please check the logs."
+    }
+    user = request.user
+
+    if not request.user.is_authenticated:
+        return Response({'Please login to view Interest Lists'},
+                        status.HTTP_401_UNAUTHORIZED)
+
+    try:
+        querySet = InterestList.objects.filter(owner=user)
+        serializer_class = InterestListSerializer(querySet, many=True)
+    except HTTPError as http_err:
+        logger.error(http_err)
+        return Response(errorMsg, status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except Exception as err:
+        logger.error(err)
+        return Response(errorMsg, status.HTTP_500_INTERNAL_SERVER_ERROR)
+    else:
+        return Response(serializer_class.data, status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+def interest_lists_subscriptions(request):
+    """Handles HTTP requests for interest lists that the request user is 
+        subscribed to"""
+    errorMsg = {
+        "message": "Error fetching records please check the logs."
```

Comment:
Is this a message that a user (ie learner) would see?  If so, it probably doesn't make sense to instruct them to check the logs.

Reply:
This method is mainly for maintainers, we show a different message on the UI.